### PR TITLE
[risk=no][RW-7280] Enable some Prettier-compatible eslint rules

### PR DESCRIPTION
**Pull request by jmthibault79**
Some rules are compatible with Prettier, some are incompatible, and some are redundant to it.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -8,8 +8,6 @@ module.exports = {
     'prettier',
     'react',
     'react-hooks',
-    'simple-import-sort',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:11]: remove for now; I have a speculative follow-on PR to re-enable.

File: ui/.eslintrc.js [hunk 2]
```diff
@@ -8,8 +8,6 @@ module.exports = {
...
-    'sort-keys-fix',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:12]: whatever this was, we no longer refer to it

File: ui/.eslintrc.js [hunk 3]
```diff
@@ -74,6 +72,9 @@ module.exports = {
     '@typescript-eslint/prefer-optional-chain': 'warn',
     '@typescript-eslint/unified-signatures': 'warn',
     // '@typescript-eslint/member-ordering': ['warn', { 'classExpressions': ['method', 'field'] }],
+    // '@typescript-eslint/prefer-function-type': 'warn',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:75]: I judged these to be more "functional" than "style" so I moved them to this section

File: ui/.eslintrc.js [hunk 4]
```diff
@@ -83,23 +84,20 @@ module.exports = {
 
     'prettier/prettier': 'warn',
 
+    'max-len': [
```
Comment (jmthibault79) [inline ui/.eslintrc.js:87]: These 2 are compatible and non-redundant to Prettier

File: ui/.eslintrc.js [hunk 5]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
+        code: 140,
+        ignorePattern: '^import |^export\\{(.*?)\\}',
+        ignoreComments: true,
+      },
+    ],
+    'no-trailing-spaces': 'warn',
+
     // temp disable others.  re-enable if desired after determining that they don't conflict with prettier.
 
-    // 'eol-last': 'warn',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:88]: redundant

File: ui/.eslintrc.js [hunk 6]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
-    // 'max-len': ['warn', {code: 140, ignorePattern: '^import |^export\\{(.*?)\\}', ignoreComments: true}],
-    // 'no-trailing-spaces': 'warn',
-    // 'quotes': ['warn', 'single'],
-    // // 'brace-style': ['warn', '1tbs'],
```
Comment (jmthibault79) [inline ui/.eslintrc.js:92]: redundant

File: ui/.eslintrc.js [hunk 7]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
-    // // 'no-multi-spaces': 'warn',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:93]: redundant

File: ui/.eslintrc.js [hunk 8]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
-    // // 'simple-import-sort/sort': 'warn',
-    // // 'space-before-function-paren': ['warn', { 'anonymous': 'never', 'named': 'never', 'asyncArrow': 'always' }],
-    // // 'spaced-comment': 'warn',
-    //
-    // '@typescript-eslint/type-annotation-spacing': 'warn',
-    // '@typescript-eslint/prefer-function-type': 'warn',
-    // // '@typescript-eslint/semi': 'warn',
-    //
-    // 'react/jsx-curly-spacing': ["warn", {'when': 'never'}],
+    // 'simple-import-sort/sort': 'warn',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:99]: deferred- these 2 are bigger changes, and maybe not desirable.

File: ui/.eslintrc.js [hunk 9]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
     // temp disable others.  re-enable if desired after determining that they don't conflict with prettier.
 
-    // 'eol-last': 'warn',
-    // 'max-len': ['warn', {code: 140, ignorePattern: '^import |^export\\{(.*?)\\}', ignoreComments: true}],
-    // 'no-trailing-spaces': 'warn',
-    // 'quotes': ['warn', 'single'],
-    // // 'brace-style': ['warn', '1tbs'],
-    // // 'no-multi-spaces': 'warn',
-    // // 'simple-import-sort/sort': 'warn',
-    // // 'space-before-function-paren': ['warn', { 'anonymous': 'never', 'named': 'never', 'asyncArrow': 'always' }],
```
Comment (jmthibault79) [inline ui/.eslintrc.js:95]: incompatible

File: ui/.eslintrc.js [hunk 10]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
-    // // 'spaced-comment': 'warn',
-    //
-    // '@typescript-eslint/type-annotation-spacing': 'warn',
```
Comment (jmthibault79) [inline ui/.eslintrc.js:98]: these are all redundant to Prettier

File: ui/.eslintrc.js [hunk 11]
```diff
@@ -83,23 +84,20 @@ module.exports = {
...
-    // '@typescript-eslint/prefer-function-type': 'warn',
-    // // '@typescript-eslint/semi': 'warn',
-    //
-    // 'react/jsx-curly-spacing': ["warn", {'when': 'never'}],
```
Comment (jmthibault79) [inline ui/.eslintrc.js:102]: redundant

File: ui/src/app/pages/analysis/runtime-panel.spec.tsx [hunk 12]
```diff
@@ -557,85 +557,96 @@ describe('RuntimePanel', () => {
     expect(runtimeApiStub.runtime.gceConfig).toBeFalsy();
   });
 
-  it('should set runtime preset values in customize panel instead of getRuntime values if configurationType is GeneralAnalysis', async () => {
-    // skip this test after enabling pd
-    if (enablePd) {
-      return;
+  it(
+    'should set runtime preset values in customize panel instead of getRuntime values ' +
```
Comment (jmthibault79) [inline ui/src/app/pages/analysis/runtime-panel.spec.tsx:561]: line was too long

File: ui/src/app/pages/analysis/runtime-panel.spec.tsx [hunk 13]
```diff
@@ -557,85 +557,96 @@ describe('RuntimePanel', () => {
...
-
-    const wrapper = await component();
-    await mustClickButton(wrapper, 'Customize');
-
-    expect(getMainCpu(wrapper)).toEqual(
-      findMachineByName(
+  );
+
+  it(
+    'should set runtime preset values in customize panel instead of getRuntime values ' +
```
Comment (jmthibault79) [inline ui/src/app/pages/analysis/runtime-panel.spec.tsx:602]: line was too long

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx [hunk 14]
```diff
@@ -429,7 +429,9 @@ describe('WorkspaceEdit', () => {
       CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID
     );
 
-    const expectedUpgradeMessage = `${CdrVersionsStubVariables.ALT_WORKSPACE_CDR_VERSION} to ${CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION}.`;
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:432]: line too long

File: ui/src/app/utils/resources.spec.tsx [hunk 15]
```diff
@@ -177,7 +177,9 @@ describe('resources.tsx', () => {
   });
 
   it('should return resource URLs', () => {
-    const WORKSPACE_URL_PREFIX = `/workspaces/${WorkspaceStubVariables.DEFAULT_WORKSPACE_NS}/${WorkspaceStubVariables.DEFAULT_WORKSPACE_ID}`;
```
Comment (jmthibault79) [inline ui/src/app/utils/resources.spec.tsx:180]: line too long

**Review by calbach — APPROVED** [review]
thanks

**Pull request merged**